client/src/app/_services/auth.service.ts

```diff
@@ -21,45 +22,25 @@ export class AuthService {
       accountType: 'Guest',
       createdAt: new Date(),
     }));
-    if (localStorage.getItem('currentUser')) {
-      const currentUser = JSON.parse(localStorage.getItem('currentUser'));
-      if (currentUser.jwt) {
-        const jwtDecoded = jwt_decode(currentUser.jwt);
-        if (Date.now() < jwtDecoded.exp * 1000) {
-          this.managePermissionAndRole(jwtDecoded.permissions);
-
-          // Update /me
-          this.me().subscribe();
-
-          // Refresh the token after 50ms to prevent other call.
-          setTimeout(
-            () => {
-              this.refresh().subscribe();
-            },
-            50,
-          );
-        } else {
-          this.clearUser();
-        }
+    const currentUser = {
+      ...JSON.parse(localStorage.getItem('currentUser')),
+      ...JSON.parse(sessionStorage.getItem('currentUser')),
+    };
+    if (currentUser.jwt) {
+      const jwtDecoded = jwt_decode(currentUser.jwt);
+      if (Date.now() < (jwtDecoded.exp * 1000 - 3600000)) { // Expiration minus 12 hours
+        this.me().subscribe();
+      } else {
+        this.clearUser();
       }
     }
   }
 
-  login(username: string, password: string): Observable<any> {
-    return this.http.post('/api/auth/login', { username, password })
-      .pipe(tap((jwt: { jwt: String }) => {
-        this.saveUser(jwt);
+  login(username: string, password: string, rememberMe: Number): Observable<any> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this 'Number' wrapper object with primitive type 'number'. ([typescript:S1533](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1533))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=279&issues=AWRkqjbIMt9UOhrw9sCt&open=AWRkqjbIMt9UOhrw9sCt)

[ISSUE_KEY]: AWRkqjbIMt9UOhrw9sCt